### DISCUSSION TITLE
Upcoming updates / Tutorials

### DISCUSSION THREAD

**Discussion by florentdemelezi** [aquacropos/aquacrop#123] [Q&A] [answered]
Dear All, 

Wanted to ask if there is a tutorial/support to run AquaCrop-OS.py in a local machine ?
I am having trouble to run as:
- Pending Deprecation in Numba 0.57.0. For more information please see: https://numba.readthedocs.io/en/stable/reference/deprecation.html#deprecation-of-the-numba-pycc-module
  from numba.pycc import CC   #package was updated
- RuntimeError: Attempted to compile AOT function without the compiler used by `numpy.distutils` present. Cannot find suitable msvc
Trying on google colab with cffi==1.15.0 or cffi==1.16.0 but both fail to simulate as there is a package mismatch 
''' # run model till termination
model.run_model(till_termination=True)''' 

As I have basic knowledge of programming I am stuck at this point on utilising AquaCrop-OS for my research.
Looking forward to be hearing form you.

Florent

**Comment by Paloschi** [comment, marked as answer]
Running AquaCrop-OS on a local machine can indeed present a few hurdles, particularly with dependencies like Numba and issues arising from attempting to compile AOT (Ahead-Of-Time) functions without the appropriate compiler tools. Here's a step-by-step guide to potentially resolve your issues, based on the error messages you've encountered:

1. Numba Pending Deprecation Warning
The warning about the deprecation of the numba.pycc module in Numba version 0.57.0 suggests that your code or one of your dependencies is using a feature that will soon be removed. While this is a warning and not an error, it's a good idea to check if there's an updated version of AquaCrop-OS or any related dependencies that might have addressed this deprecation.

2. Runtime Error: Missing Compiler
The RuntimeError regarding the absence of a suitable MSVC (Microsoft Visual C++) compiler is more critical. AquaCrop-OS's use of Numba for AOT compilation requires a C compiler to be available:

- For Windows: Ensure that you have the Microsoft C++ Build Tools installed. You can download them from the Visual Studio downloads page (look for "Build Tools for Visual Studio"). After installation, you might need to restart your command prompt or IDE to ensure the environment variables are updated.

- For Linux/Mac: Install GCC (GNU Compiler Collection) if it's not already present. This can usually be done through your package manager (apt for Ubuntu, brew for macOS, etc.).

3. Google Colab Issues with cffi
The problems with package mismatches in Google Colab suggest that there might be conflicting dependencies. The cffi package is often used for interfacing Python code with C, and mismatched versions can indeed cause issues. To potentially resolve this:
- Ensure that all dependencies are compatible with each other. This might involve installing specific versions of certain packages. You can use pip commands with version specifications, e.g., pip install cffi==1.14.0 if you find that version 1.15.0 or 1.16.0 cause issues.
- Google Colab environments can be reset to ensure a clean state; this might help if there are lingering issues from previous installations.

General Steps to Run AquaCrop-OS Locally:

- Create a Virtual Environment: This is a good practice to avoid conflicts between project dependencies. Use python -m venv aquacrop-env to create and source aquacrop-env/bin/activate (Linux/Mac) or aquacrop-env\Scripts\activate (Windows) to activate it.

- Install Dependencies: Within your virtual environment, install the required packages, including specific versions that you know are compatible. The pip tool is your friend here.

- Compiler Setup: Ensure that the appropriate compiler is installed and accessible in your environment's path, as discussed above.

Cheers

**Reply by anj-ML** [reply]
ImportError Traceback (most recent call last)
Cell In[1], line 1
----> 1 from aquacrop import AquaCropModel

ImportError: cannot import name 'AquaCropModel' from 'aquacrop' (/home/dr-anjum/anaconda3/lib/python3.12/site-packages/aquacrop/init.py)

I have installed aqua crop os-py , mnay times correctly , but it does not work .I got above messgae. Please help out .
Thanks in adavance

**Reply by anj-ML** [reply]
no module  named aquacrop

**Comment by tfoster88** [comment]
Thanks for that really detailed and useful answer @Paloschi. Just to add that we are going to look into implementing an alternative to numba AOT over the coming months, but in the meantime the temporary fix if none of the above work for you is to run in development environment which turns off the compilation. It will be slower but will usually eliminate issues that are traced back to the numba compilation, which we will look into a more complete solution to as its clearly becoming a more common problem

**Comment by lopezjuana** [comment]
Good morning: I am working on a project and I am interested in being able to use aquacrop to simulate crop growth and irrigation needs in an irrigation district. I want to develop a tool (web or desktop) that connects to a database to obtain the data that aquacrop requires to be able to perform the simulations.
The question is if you already have the alternative to numba AOT, since we are now testing aquacrop in development mode. Is there any experience of implementation of aquacrop in tools to provide advice to producers?
I am writing from Santiago del Estero, Argentina.

Greetings, Juana
